llvm/lib/Target/NVPTX/NVPTXAsmPrinter.cpp

```diff
@@ -914,7 +914,7 @@ void NVPTXAsmPrinter::emitLinkageDirective(const GlobalValue *V,
       else
         O << ".visible ";
     } else if (V->hasAppendingLinkage()) {
-      report_fatal_error("Symbol '" + llvm::Twine(V->getNameOrAsOperand()) +
+      report_fatal_error("Symbol '" + (V->hasName() ? V->getName() : "") +
```

Comment:
IMO, it would make sense to make `getNameOrAsOperand()` available in general, but that's unrelated to your patch.

I think the initial commit that introduced `getNameOrAsOperand` was somewhat misguided to put that function under NDEBUG. IMO debug/nodebug should not change availability of class members. It's OK to change member function implementation, if necessary. E.g how it's done here:
https://github.com/llvm/llvm-project/blob/c2e96778e04197dd266f7c540bf174b6ec28a434/llvm/include/llvm/IR/Value.h#L338-L342
It just adds unnecessary headache without buying us anything.

That said, that `getNameOrAsOperand` is a terrible name. Your code as is makes more sense. Let's keep it.